src/aave-v3/SupplyVault.sol

```diff
@@ -126,37 +122,40 @@ contract SupplyVault is ISupplyVault, SupplyVaultBase {
         view
         returns (address[] memory rewardTokens, uint256[] memory unclaimedAmounts)
     {
-        uint256 supply = totalSupply();
-        if (supply > 0) {
-            uint256[] memory claimableAmounts;
+        uint256[] memory claimableAmounts;
 
-            {
-                address[] memory poolTokens = new address[](1);
-                poolTokens[0] = poolToken;
+        {
+            address[] memory poolTokens = new address[](1);
+            poolTokens[0] = poolToken;
 
-                (rewardTokens, claimableAmounts) = rewardsManager.getAllUserRewards(
-                    poolTokens,
-                    address(this)
-                );
-            }
+            (rewardTokens, claimableAmounts) = rewardsManager.getAllUserRewards(
+                poolTokens,
+                address(this)
+            );
+        }
+
+        unclaimedAmounts = new uint256[](claimableAmounts.length);
+        uint256 supply = totalSupply();
 
-            for (uint256 i; i < rewardTokens.length; ) {
-                address rewardToken = rewardTokens[i];
-                UserRewardsData memory userRewardsData = userRewards[rewardToken][_user];
+        for (uint256 i; i < rewardTokens.length; ++i) {
+            address rewardToken = rewardTokens[i];
+            UserRewardsData memory userRewardsData = userRewards[rewardToken][_user];
 
+            if (supply > 0)
                 unclaimedAmounts[i] =
                     userRewardsData.unclaimed +
                     balanceOf(_user).mulDivDown(
                         (rewardsIndex[rewardToken] +
-                            claimableAmounts[i].mulDivDown(SCALE, totalSupply())) -
-                            userRewardsData.index,
+                            claimableAmounts[i].mulDivDown(SCALE, supply)) - userRewardsData.index,
+                        SCALE
+                    );
```

Comment:
Perhaps extract this to an internal view function `_computeUserUnclaimedRewards(address _user, address _rewardToken, uint256 _claimableAmount)`

We'd no longer benefit from storing `totalSupply()` in the stack but code would be reused in `getUnclaimedRewards`...

Reply:
I was thinking about doing it in another PR but can be done here if it's a blocker

Reply:
second thought, I won't do it in this PR, this is better to do it in another PR and discuss about this specific.

Btw we can pass `_supply` as an argument and still benefit from the memory variable

Reply:
https://github.com/morpho-dao/morpho-tokenized-vaults/pull/191/files

I'm working on this PR to extract this out. I will rebase it onto this PR.